File: day1/day1.rs
use std::fs::File;
use std::io::{BufRead, BufReader};

fn main() {
    let file = File::open("input.txt").unwrap();
    let reader = BufReader::new(file);

    let mut numbers = vec![];
    let mut tmp = 0;
    for line in reader.lines() {
        let line = line.unwrap();
        if !line.is_empty() {
            let number: i32 = line.parse().unwrap_or_default();
            tmp += number;
        } 
        else {
            numbers.push(tmp);
            tmp = 0;
        }
    };

    numbers.sort();

    let answer1 = numbers[numbers.len() - 1];
    let answer2: i32 = numbers.iter().rev().take(3).sum();

    //println!("Numbers in file: {:#?}", numbers);
    println!("{:?}", answer1);
    println!("{}", answer2);
}

File: day01/day1.py
f = open("input.txt", "r")
lines = f.readlines()
lines = [x.strip('\n') for x in lines]

cal = []

tmp = []
for x in lines:
    if x != "":
        tmp.append(int(x))
    else:
        cal.append(tmp)
        tmp = []

# part 1
print(max([sum(x) for x in cal]))

# part 2
print(sum(sorted([sum(x) for x in cal])[-3:]))